runtime/compiler/control/CompilationThread.cpp

```diff
@@ -7139,6 +7162,43 @@ TR::CompilationInfoPerThreadBase::exitPerClientAllocationRegion()
       _compiler->switchToGlobalMemory();
    _perClientPersistentMemory = NULL;
    }
+
+void
+TR::CompilationInfoPerThreadBase::downgradeLocalCompilationIfLowPhysicalMemory(TR_MethodToBeCompiled *entry)
+   {
+   TR_ASSERT_FATAL(_compInfo.getPersistentInfo()->getRemoteCompilationMode() == JITServer::CLIENT,
+                   "Must be called on JITServer client");
+
+   J9Method *method = entry->getMethodDetails().getMethod();
+   if (!TR::Options::getCmdLineOptions()->getOption(TR_DisableJITServerBufferedExpensiveCompilations) &&
+       TR::Options::getCmdLineOptions()->allowRecompilation() &&
+       !TR::CompilationInfo::isCompiled(method) &&  // do not downgrade recompilations
+       entry->_optimizationPlan->getOptLevel() >= warm ||
```

Comment:
We need another parenthesis here, otherwise `TR_DisableJITServerBufferedExpensiveCompilations` is going to be ignored for AOT compilations.
```
(entry->_optimizationPlan->getOptLevel() >= warm ||
// AOT compilations that are upgraded back to cheap-warm should be considered expensive
       (entry->_useAotCompilation && !TR::Options::getAOTCmdLineOptions()->getOption(TR_DisableAotAtCheapWarm))))
```

Reply:
fixed.